core/src/main/java/org/apache/spark/shuffle/api/ShuffleDataIO.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.shuffle.api;
```

Comment:
I'm concerned about conflicts with the other kinds of APIs in the `org.apache.spark.api.*` namespace, particularly because these are all related to other language bindings, e.g. `org.apache.spark.api.java.function.Function`, `org.apache.spark.api.r.RRDD`. Let's modify `SparkBuild.scala` instead - I'll look into what that would require. Can that be done in a follow-up PR?

Reply:
Sure. Better be proactive and file a bug to make these interfaces non-Private and at the same time make sure they're showing up properly in documentation.

Reply:
https://issues.apache.org/jira/browse/SPARK-28568